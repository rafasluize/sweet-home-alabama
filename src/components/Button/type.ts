export type TVariant = 'add' | 'remove';
