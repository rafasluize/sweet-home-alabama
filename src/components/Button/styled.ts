import styled, { css } from 'styled-components';
import { TVariant } from './type';

export const Button = styled.button<{ variant: TVariant }>`
  ${({ variant }) =>
    variant &&
    variant === 'add' &&
    css`
      background: #40d9b8;
      img {
        width: 24px;
        height: 24px;
      }
      border-radius: 100px;
      width: 48px;
      height: 48px;
    `}
  border: none;
`;

export default Button;
