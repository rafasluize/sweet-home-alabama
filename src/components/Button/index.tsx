import React, { FC } from 'react';
import add from '../../assets/icons/ic-add.png';
import remove from '../../assets/icons/ic-delete.png';
import * as Styled from './styled';
import { TVariant } from './type';

const Button: FC<{ variant: TVariant; onClick: () => void }> = ({ variant, onClick }) => (
  <Styled.Button variant={variant} type="button" onClick={onClick}>
    {variant === 'add' && <img src={add} alt="Adicionar item" />}
    {variant === 'remove' && <img src={remove} alt="Remover item" />}
  </Styled.Button>
);

export default Button;
