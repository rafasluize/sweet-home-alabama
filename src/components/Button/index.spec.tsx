import React from 'react';
import { render, screen } from '@testing-library/react';
import Button from '.';

const handle = () => false;

jest.mock('../../assets/icons/ic-add.png');

describe('Button', () => {
  it('should img add', () => {
    render(<Button variant="add" onClick={handle} />);

    expect(screen.getByAltText(/Adicionar item/i)).toBeDefined();
  });

  it('snapshot', () => {
    const { container } = render(<Button variant="remove" onClick={handle} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
