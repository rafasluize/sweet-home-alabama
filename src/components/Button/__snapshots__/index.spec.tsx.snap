// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
.c0 {
  border: none;
}

<button
  class="c0"
  data-testid="button-remove"
  role="button"
  type="button"
>
  <img
    alt="Remover item"
    src="test"
  />
</button>
`;
